### PR TITLE
Add support for Interlogix Smoke Detector and Pet Immune SAW PIR Motion Sensor

### PR DESCRIPTION
**Pull request by yannicklarrivee**

### CODE REVIEW

**Comment by zuckschwerdt** [comment]
Thanks. This confirms #1708.

File: src/devices/interlogix.c [hunk 1]
```diff
@@ -186,6 +187,14 @@ static int interlogix_decode(r_device *decoder, bitbuffer_t *bitbuffer)
         f3_latch_state = ((message[3] & 0xe) == 0xc) ? "CLOSED" : "OPEN";
         f4_latch_state = ((message[3] & 0xe) == 0x2) ? "CLOSED" : "OPEN";
         f5_latch_state = ((message[3] & 0xe) == 0xa) ? "CLOSED" : "OPEN";
+    // PET Immune SAW PIR motion sensor logic.
+    } else if ((reverse8(message[2]) >> 4) == 0x4) {
+        low_battery    = (message[3] & 0x10) ? "LOW" : "OK";
```
Comment (zuckschwerdt) [inline src/devices/interlogix.c:192]: Use `low_battery    = (message[3] & 0x10) ? 1 : 0;` (same as in the block below)

Comment (yannicklarrivee) [reply src/devices/interlogix.c]: Change made. Thanks for the review.

**Comment by zuckschwerdt** [comment]
@yannicklarrivee this change causes a regression with the test samples tests/interlogix/03 which state "03-Door contact sensor #1 - trip by opening door".

Can you take a look how those were wrongly identified as SAW PIR now?

S.a. https://github.com/merbanan/rtl_433_tests/tree/master/tests/interlogix
and https://github.com/merbanan/rtl_433_tests/issues/446

**Comment by merbanan** [comment]
((message[3] & 0x04) | ((message[3] & 0x08) == 0x8)) <-- that looks bogus. Should it not be || ?

**Comment by zuckschwerdt** [comment]
Yes, looks strange. Should be `(message[3] & 0x0a)` to test for any, or maybe `((message[3] & 0x04) || (message[3] & 0x08))` to (xor) test for either.
Sorry I merged this. I didn't spot it was just that 0x8 botched in. A detailed description what/why is needed.

**Comment by yannicklarrivee** [comment]
Tests are not failing for Interlogix when I run it.
`python3 bin/run_test.py -I time --first-line
WARNING: Ignoring 'tests/honeywell/RE208/g014_344.975M_250k.cu8'
WARNING: Ignoring 'tests/honeywell/RE208/g012_344.975M_250k.cu8'
WARNING: Ignoring 'tests/honeywell/RE208/g011_344.975M_250k.cu8'
WARNING: Ignoring 'tests/honeywell/RE208/g013_344.975M_250k.cu8'
WARNING: Ignoring 'tests/honeywell/RE208/g015_344.975M_250k.cu8'
WARNING: Ignoring 'tests/honeywell/2Gig-DW10/g007_344.975M_250k.cu8'
WARNING: Ignoring 'tests/honeywell/2Gig-DW10/g004_344.975M_250k.cu8'
WARNING: Ignoring 'tests/honeywell/2Gig-DW10/g009_344.975M_250k.cu8'
WARNING: Ignoring 'tests/honeywell/2Gig-DW10/g002_344.975M_250k.cu8'
WARNING: Ignoring 'tests/honeywell/2Gig-DW10/g001_344.975M_250k.cu8'
WARNING: Ignoring 'tests/honeywell/2Gig-DW10/g010_344.975M_250k.cu8'
WARNING: Ignoring 'tests/honeywell/2Gig-DW10/g003_344.975M_250k.cu8'
WARNING: Ignoring 'tests/honeywell/2Gig-DW10/g005_344.975M_250k.cu8'
WARNING: Ignoring 'tests/honeywell/2Gig-DW10/g008_344.975M_250k.cu8'
WARNING: Ignoring 'tests/honeywell/2Gig-DW10/g006_344.975M_250k.cu8'
WARNING: Missing 'tests/idm/IDM/g001_912.6M_2359.3k.cu8'
WARNING: Ignoring 'tests/idm/NetIDM/g012_912.6M_2359.3k.cu8'
WARNING: Ignoring 'tests/idm/NetIDM/g011_912.6M_2359.3k.cu8'
WARNING: Ignoring 'tests/idm/NetIDM/g010_912.6M_2359.3k.cu8'
WARNING: Ignoring 'tests/wec_1502/01/gfile002.cu8'
WARNING: Ignoring 'tests/scmplus/01/g005_912.6M_2359.3k.cu8'
WARNING: Ignoring 'tests/scmplus/01/g003_912.6M_2359.3k.cu8'
WARNING: Ignoring 'tests/scmplus/01/g002_912.6M_2359.3k.cu8'
WARNING: Ignoring 'tests/inkbird/inkbird_ith20r/g001_433.92M_250k.cu8'
WARNING: Ignoring 'tests/inkbird/inkbird_ith20r/g002_433.92M_250k.cu8'
WARNING: Ignoring 'tests/inkbird/inkbird_ith20r/g003_433.92M_250k.cu8'
WARNING: Ignoring 'tests/inkbird/IBS-P01R/g071_433.92M_250k.cu8'
WARNING: Ignoring 'tests/inkbird/IBS-P01R/g026_433.92M_250k.cu8'
WARNING: Ignoring 'tests/inkbird/IBS-P01R/g001_433.92M_250k.cu8'
## Fail with 'tests/cotech/cotech_36-7959/01/g013_433.92M_1000k.cu8':
 dictionary_item_added
  * root['light_lux']
  * root['uv']
 Expected: {'model': 'Cotech-367959', 'id': 84, 'battery_ok': 1, 'temperature_F': 73.0, 'humidity': 34, 'rain_mm': 79.2, 'wind_dir_deg': 190, 'wind_avg_m_s': 0.6, 'wind_max_m_s': 2.0, 'mic': 'CRC'}
  But got: {'model': 'Cotech-367959', 'id': 84, 'battery_ok': 1, 'temperature_F': 73.0, 'humidity': 34, 'rain_mm': 79.2, 'wind_dir_deg': 190, 'wind_avg_m_s': 0.6, 'wind_max_m_s': 2.0, 'light_lux': 69627, 'uv': 251, 'mic': 'CRC'}
## Fail with 'tests/cotech/cotech_36-7959/01/g020_433.92M_1000k.cu8':
 dictionary_item_added
  * root['light_lux']
  * root['uv']
 Expected: {'model': 'Cotech-367959', 'id': 84, 'battery_ok': 1, 'temperature_F': 73.3, 'humidity': 34, 'rain_mm': 79.2, 'wind_dir_deg': 241, 'wind_avg_m_s': 0.0, 'wind_max_m_s': 0.0, 'mic': 'CRC'}
  But got: {'model': 'Cotech-367959', 'id': 84, 'battery_ok': 1, 'temperature_F': 73.3, 'humidity': 34, 'rain_mm': 79.2, 'wind_dir_deg': 241, 'wind_avg_m_s': 0.0, 'wind_max_m_s': 0.0, 'light_lux': 69627, 'uv': 251, 'mic': 'CRC'}
## Fail with 'tests/cotech/cotech_36-7959/01/g026_433.92M_1000k.cu8':
 dictionary_item_added
  * root['light_lux']
  * root['uv']
 Expected: {'model': 'Cotech-367959', 'id': 84, 'battery_ok': 1, 'temperature_F': 73.3, 'humidity': 34, 'rain_mm': 79.2, 'wind_dir_deg': 126, 'wind_avg_m_s': 0.0, 'wind_max_m_s': 0.0, 'mic': 'CRC'}
  But got: {'model': 'Cotech-367959', 'id': 84, 'battery_ok': 1, 'temperature_F': 73.3, 'humidity': 34, 'rain_mm': 79.2, 'wind_dir_deg': 126, 'wind_avg_m_s': 0.0, 'wind_max_m_s': 0.0, 'light_lux': 69627, 'uv': 251, 'mic': 'CRC'}
## Fail with 'tests/cotech/cotech_36-7959/01/g005_433.92M_1000k.cu8':
 dictionary_item_added
  * root['light_lux']
  * root['uv']
 Expected: {'model': 'Cotech-367959', 'id': 84, 'battery_ok': 1, 'temperature_F': 73.0, 'humidity': 34, 'rain_mm': 79.2, 'wind_dir_deg': 355, 'wind_avg_m_s': 0.0, 'wind_max_m_s': 0.0, 'mic': 'CRC'}
  But got: {'model': 'Cotech-367959', 'id': 84, 'battery_ok': 1, 'temperature_F': 73.0, 'humidity': 34, 'rain_mm': 79.2, 'wind_dir_deg': 355, 'wind_avg_m_s': 0.0, 'wind_max_m_s': 0.0, 'light_lux': 69627, 'uv': 251, 'mic': 'CRC'}
## Fail with 'tests/cotech/cotech_36-7959/01/g015_433.92M_1000k.cu8':
 dictionary_item_added
  * root['light_lux']
  * root['uv']
 Expected: {'model': 'Cotech-367959', 'id': 84, 'battery_ok': 1, 'temperature_F': 73.0, 'humidity': 34, 'rain_mm': 79.2, 'wind_dir_deg': 188, 'wind_avg_m_s': 1.1, 'wind_max_m_s': 1.7, 'mic': 'CRC'}
  But got: {'model': 'Cotech-367959', 'id': 84, 'battery_ok': 1, 'temperature_F': 73.0, 'humidity': 34, 'rain_mm': 79.2, 'wind_dir_deg': 188, 'wind_avg_m_s': 1.1, 'wind_max_m_s': 1.7, 'light_lux': 69627, 'uv': 251, 'mic': 'CRC'}
## Fail with 'tests/cotech/cotech_36-7959/01/g009_433.92M_1000k.cu8':
 dictionary_item_added
  * root['light_lux']
  * root['uv']
 Expected: {'model': 'Cotech-367959', 'id': 84, 'battery_ok': 1, 'temperature_F': 73.1, 'humidity': 34, 'rain_mm': 79.2, 'wind_dir_deg': 91, 'wind_avg_m_s': 0.0, 'wind_max_m_s': 0.0, 'mic': 'CRC'}
  But got: {'model': 'Cotech-367959', 'id': 84, 'battery_ok': 1, 'temperature_F': 73.1, 'humidity': 34, 'rain_mm': 79.2, 'wind_dir_deg': 91, 'wind_avg_m_s': 0.0, 'wind_max_m_s': 0.0, 'light_lux': 69627, 'uv': 251, 'mic': 'CRC'}
## Fail with 'tests/cotech/cotech_36-7959/01/g024_433.92M_1000k.cu8':
 dictionary_item_added
  * root['light_lux']
  * root['uv']
 Expected: {'model': 'Cotech-367959', 'id': 84, 'battery_ok': 1, 'temperature_F': 73.3, 'humidity': 34, 'rain_mm': 79.2, 'wind_dir_deg': 57, 'wind_avg_m_s': 0.0, 'wind_max_m_s': 0.0, 'mic': 'CRC'}
  But got: {'model': 'Cotech-367959', 'id': 84, 'battery_ok': 1, 'temperature_F': 73.3, 'humidity': 34, 'rain_mm': 79.2, 'wind_dir_deg': 57, 'wind_avg_m_s': 0.0, 'wind_max_m_s': 0.0, 'light_lux': 69627, 'uv': 251, 'mic': 'CRC'}
## Fail with 'tests/cotech/cotech_36-7959/01/g029_433.92M_1000k.cu8':
 dictionary_item_added
  * root['light_lux']
  * root['uv']
 Expected: {'model': 'Cotech-367959', 'id': 84, 'battery_ok': 1, 'temperature_F': 73.3, 'humidity': 34, 'rain_mm': 84.0, 'wind_dir_deg': 191, 'wind_avg_m_s': 0.1, 'wind_max_m_s': 0.7, 'mic': 'CRC'}
  But got: {'model': 'Cotech-367959', 'id': 84, 'battery_ok': 1, 'temperature_F': 73.3, 'humidity': 34, 'rain_mm': 84.0, 'wind_dir_deg': 191, 'wind_avg_m_s': 0.1, 'wind_max_m_s': 0.7, 'light_lux': 69627, 'uv': 251, 'mic': 'CRC'}
## Fail with 'tests/cotech/cotech_36-7959/01/g007_433.92M_1000k.cu8':
 dictionary_item_added
  * root['light_lux']
  * root['uv']
 Expected: {'model': 'Cotech-367959', 'id': 84, 'battery_ok': 1, 'temperature_F': 73.0, 'humidity': 34, 'rain_mm': 79.2, 'wind_dir_deg': 269, 'wind_avg_m_s': 0.0, 'wind_max_m_s': 0.0, 'mic': 'CRC'}
  But got: {'model': 'Cotech-367959', 'id': 84, 'battery_ok': 1, 'temperature_F': 73.0, 'humidity': 34, 'rain_mm': 79.2, 'wind_dir_deg': 269, 'wind_avg_m_s': 0.0, 'wind_max_m_s': 0.0, 'light_lux': 69627, 'uv': 251, 'mic': 'CRC'}
## Fail with 'tests/cotech/cotech_36-7959/01/g001_433.92M_1000k.cu8':
 dictionary_item_added
  * root['light_lux']
  * root['uv']
 Expected: {'model': 'Cotech-367959', 'id': 84, 'battery_ok': 1, 'temperature_F': 73.1, 'humidity': 34, 'rain_mm': 79.2, 'wind_dir_deg': 188, 'wind_avg_m_s': 0.0, 'wind_max_m_s': 0.0, 'mic': 'CRC'}
  But got: {'model': 'Cotech-367959', 'id': 84, 'battery_ok': 1, 'temperature_F': 73.1, 'humidity': 34, 'rain_mm': 79.2, 'wind_dir_deg': 188, 'wind_avg_m_s': 0.0, 'wind_max_m_s': 0.0, 'light_lux': 69627, 'uv': 251, 'mic': 'CRC'}
## Fail with 'tests/cotech/cotech_36-7959/01/g022_433.92M_1000k.cu8':
 dictionary_item_added
  * root['light_lux']
  * root['uv']
 Expected: {'model': 'Cotech-367959', 'id': 84, 'battery_ok': 1, 'temperature_F': 73.3, 'humidity': 34, 'rain_mm': 79.2, 'wind_dir_deg': 299, 'wind_avg_m_s': 0.0, 'wind_max_m_s': 0.0, 'mic': 'CRC'}
  But got: {'model': 'Cotech-367959', 'id': 84, 'battery_ok': 1, 'temperature_F': 73.3, 'humidity': 34, 'rain_mm': 79.2, 'wind_dir_deg': 299, 'wind_avg_m_s': 0.0, 'wind_max_m_s': 0.0, 'light_lux': 69627, 'uv': 251, 'mic': 'CRC'}
## Fail with 'tests/cotech/cotech_36-7959/01/g002_433.92M_1000k.cu8':
 dictionary_item_added
  * root['light_lux']
  * root['uv']
 Expected: {'model': 'Cotech-367959', 'id': 84, 'battery_ok': 1, 'temperature_F': 73.1, 'humidity': 34, 'rain_mm': 79.2, 'wind_dir_deg': 189, 'wind_avg_m_s': 0.0, 'wind_max_m_s': 0.0, 'mic': 'CRC'}
  But got: {'model': 'Cotech-367959', 'id': 84, 'battery_ok': 1, 'temperature_F': 73.1, 'humidity': 34, 'rain_mm': 79.2, 'wind_dir_deg': 189, 'wind_avg_m_s': 0.0, 'wind_max_m_s': 0.0, 'light_lux': 69627, 'uv': 251, 'mic': 'CRC'}
WARNING: Ignoring 'tests/adlm_fprf/01/g013_433.92M_250k.cu8'
WARNING: Ignoring 'tests/adlm_fprf/01/g014_433.92M_250k.cu8'
WARNING: Ignoring 'tests/adlm_fprf/01/g011_433.92M_250k.cu8'
WARNING: Ignoring 'tests/adlm_fprf/01/g009_433.92M_250k.cu8'
WARNING: Ignoring 'tests/adlm_fprf/01/g005_433.92M_250k.cu8'
WARNING: Ignoring 'tests/adlm_fprf/01/g007_433.92M_250k.cu8'
WARNING: Ignoring 'tests/adlm_fprf/01/g003_433.92M_250k.cu8'
WARNING: Ignoring 'tests/IKEA_Sparsnas/01/g004_867.95M_250k.cu8'
WARNING: Ignoring 'tests/IKEA_Sparsnas/01/g001_867.95M_250k.cu8'
WARNING: Ignoring 'tests/IKEA_Sparsnas/01/g005_867.95M_250k.cu8'
WARNING: Ignoring 'tests/IKEA_Sparsnas/01/g006_867.95M_250k.cu8'
WARNING: Ignoring 'tests/IKEA_Sparsnas/01/g003_867.95M_250k.cu8'
WARNING: Ignoring 'tests/IKEA_Sparsnas/01/g007_867.95M_250k.cu8'
WARNING: Ignoring 'tests/IKEA_Sparsnas/01/g002_867.95M_250k.cu8'
WARNING: Ignoring 'tests/IKEA_Sparsnas/01/g009_867.95M_250k.cu8'
WARNING: Ignoring 'tests/IKEA_Sparsnas/01/g010_867.95M_250k.cu8'
WARNING: Ignoring 'tests/IKEA_Sparsnas/01/g008_867.95M_250k.cu8'
WARNING: KlikAanKlikUit-Switch generated 45 false positive(s) in other decoders: {'Proove-Security'}
WARNING: Nexa-Security generated 125 false positive(s) in other decoders: {'KlikAanKlikUit-Switch', 'Proove-Security'}
WARNING: LaCrosse-TX35DTHIT generated 1 false positive(s) in other decoders: {'LaCrosse-TX29IT'}
WARNING: NETIDM generated 1 false positive(s) in other decoders: {'IDM'}
WARNING: Proove-Security generated 67 false positive(s) in other decoders: {'KlikAanKlikUit-Switch', 'Nexa-Security'}
WARNING: Acurite-609TXC generated 6 false positive(s) in other decoders: {'Oregon-SL109H'}
WARNING: Akhan-100F14 generated 193 false positive(s) in other decoders: {'Waveman-Switch'}
WARNING: Rubicson-Temperature generated 5 false positive(s) in other decoders: {'Solight-TE44'}
WARNING: Abarth-124Spider generated 1 false positive(s) in other decoders: {'Renault'}
WARNING: LaCrosse-R3 generated 4 false positive(s) in other decoders: {'LaCrosse-BreezePro'}
WARNING: LaCrosse-R1 generated 2 false positive(s) in other decoders: {'LaCrosse-TH3'}
WARNING: Fineoffset-WH0530 generated 1 false positive(s) in other decoders: {'GT-WT02'}
1819 records tested, 12 have failed
make: *** [test] Error 12`

**Comment by zuckschwerdt** [comment]
*Works for me* does not help.
- Why does this code (`== 0x4`) match an old sample that is now SAW PIR?
- What was that `0x08` change supposed to do?
- Any documentation on the overall idea behind this change?

We should maybe revert, I can't see that this change works by more than accident.

**Comment by zuckschwerdt** [comment]
@yannicklarrivee can you help fix the issues by providing some documentation, analysis results or the basic idea to this change?

**Comment by zuckschwerdt** [comment]
Reverted until we work this out in detail and with confidence it is working as intended.

Just let us know what results you saw on your tests and how this was intended, then we can help out to make the code robust.

**Comment by NasaGeek** [comment]
I think this changed should be re-introduced. The change in test behavior that it brought about is actually exactly what we would have expected (switch1 flipped). For some reason the test's readme appears incorrect. Interlogix test 3 was testing a motion sensor from the beginning, not a contact sensor; that was not a regression introduced by this PR.

**Comment by zuckschwerdt** [comment]
Can you document why `f1_latch_state` for PIR is different, are we actually reading the debounced level there? And why `or` both values?

**Comment by gdt** [comment]
The idea of "stop supporting legacy" doesn't make sense.   (really, I don't like any use of the word legacy :-)   But seriously, the question is whether people are likely to have them and have them functioning.    if it's hard to believe any devices are out there and working, that's one thing, but that seems like a suprising and unlikely situation.

I would ask:  is there really no type bit?   How is the intended receiver deciding how to interpret things?    Or is it just that the protocol has different kinds of alarm bits, and we should represent all of them?

**Comment by yannicklarrivee** [comment]
I totally get your point, so that’s why I thought a new dive might be
better. I didn’t find anything in the message receive that could help
differentiate those devices as they have the same type. Receiver is
programmed accordingly and they don’t use those two type of devices in a
single installation. I find it really that they programmed those new
devices this way.

**Pull request merged**